src/libraries/xsync/src/xsync_handler.cpp

```diff
@@ -183,6 +183,9 @@ void xsync_handler_t::get_blocks(uint32_t msg_size, const vnetwork::xvnode_addre
     std::vector<xblock_ptr_t> vector_blocks;
     for (uint32_t height = start_height, i = 0; height < start_height + count && i < max_request_block_count; height++) {
         auto blocks = m_sync_store->load_block_objects(owner, height);
+        if (blocks.empty()) {
+            break;
+        }
```

Comment:
missing block will be break?

Reply:
empty means local does not has block at specific height, it is reasonable to stop reading higher blocks.
other peers will request higher blocks in coming requests.